engines/bops_api/app/controllers/concerns/bops_api/error_handler.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module BopsApi
+  module ErrorHandler
+    extend ActiveSupport::Concern
+
+    EXCEPTIONS = {
+      "ActiveRecord::RecordInvalid" => {status: 400, message: "Record Invalid"},
+      "InvalidSchemaError" => {status: 400, message: "Bad Request"},
+      "NotPermittedError" => {status: 403, message: "Forbidden"},
+      "ActiveRecord::RecordNotFound" => {status: 404, message: "Cannot find record"},
+      "InternalServerError" => {status: 500, message: "Internal Server Error"}
+    }.freeze
+
+    class BaseError < StandardError; end
+
+    EXCEPTIONS.each_key do |exception_name|
+      const_set(exception_name, Class.new(BaseError)) unless const_defined?(exception_name)
+    end
+
+    included do
+      # Fallback rescue handler
+      rescue_from StandardError do |exception|
+        handle_error(exception, EXCEPTIONS["InternalServerError"])
+      end
+
+      EXCEPTIONS.each do |exception_name, context|
+        exception_class = const_get(exception_name)
+        rescue_from exception_class do |exception|
+          handle_error(exception, context)
+        end
+      end
```

Comment:
There's no need to add multiple `rescue_from` bocks - just rescue `StandardError` and the mapping will do the rest - if there's no mapping fallback to 500 error

Reply:
Ahh yes because we inherit from this anyway